src/main/java/io/divolte/server/Server.java

```diff
@@ -61,29 +66,90 @@ public Server(final ValidatedConfiguration vc) {
+                  .stream()
+                  .filter(sink -> referencedSinkNames.contains(sink.getKey()))
+                  .filter(sink -> vc.configuration().global.hdfs.enabled || !(sink.getValue() instanceof HdfsSinkConfiguration))
+                  .filter(sink -> vc.configuration().global.kafka.enabled || !(sink.getValue() instanceof KafkaSinkConfiguration))
+                  .<Map.Entry<String,ProcessingPool<?, AvroRecordBuffer>>>map(sink ->
+                          Maps.immutableEntry(sink.getKey(),
+                                              sink.getValue()
+                                                  .getFactory()
+                                                  .create(vc, sink.getKey(), schemaRegistry)))
+                  .collect(MoreCollectors.toImmutableMap());
+        logger.info("Initialized sinks: {}", sinks.keySet());
+
+        logger.debug("Initializing mappings...");
+        final Function<String, Optional<ProcessingPool<?, AvroRecordBuffer>>> schemaProvider =
+                sinkName -> Optional.ofNullable(sinks.get(sinkName));
+        mappingProcessors =
+                ImmutableMap.copyOf(Maps.transformEntries(vc.configuration().mappings,
+                                                          (mappingName, config) ->
+                                                                  new IncomingRequestProcessingPool(vc,
+                                                                                                    mappingName,
+                                                                                                    schemaRegistry,
+                                                                                                    schemaProvider,
+                                                                                                    listener)));
+        logger.info("Initialized mappings: {}", mappingProcessors.keySet());
+
+        logger.debug("Initializing sources...");
+        // First build a list of which mappings are used by each source.
+        final ImmutableMultimap<String,IncomingRequestProcessingPool> mappingProcessorsBySource =
+            vc.configuration().mappings.entrySet()
+              .stream()
+              .flatMap(mappingConfig -> {
+                  final IncomingRequestProcessingPool mappingProcessor = mappingProcessors.get(mappingConfig.getKey());
+                  return mappingConfig.getValue()
+                                      .sources
+                                      .stream()
+                                      .map(source -> Maps.immutableEntry(source, mappingProcessor));
+              })
+              .collect(MoreCollectors.toImmutableMultimap());
+        // Now instantiate all the sources. We do this in parallel because instantiation can be quite slow.
+        final ImmutableMap<String, BrowserSource> sources =
+                vc.configuration().sources.keySet()
+                  .parallelStream()
```

Comment:
So, it does actually run it in parallel using it's own ForkJoin pool. I added this because the JavaScript minification happens here, and that takes several seconds. This was a simple attempt to get the tests to run faster. It had limited impact because the Closure Compiler already does as much as it can in parallel, but was still a win.